cmd/slsa-provenance/cli/github-release_test.go

```diff
@@ -0,0 +1,94 @@
+package cli_test
+
+import (
+	"context"
+	"os"
+	"path"
+	"runtime"
+	"testing"
+
+	gh "github.com/google/go-github/v39/github"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/philips-labs/slsa-provenance-action/cmd/slsa-provenance/cli"
+	"github.com/philips-labs/slsa-provenance-action/lib/github"
+)
+
+func TestProvenenaceGitHubRelease(t *testing.T) {
+	githubToken := os.Getenv("GITHUB_TOKEN")
+	if githubToken == "" {
+		t.Skip("skipping as GITHUB_TOKEN environment variable isn't set")
+	}
+	assert := assert.New(t)
+
+	_, filename, _, _ := runtime.Caller(0)
+	rootDir := path.Join(path.Dir(filename), "../../..")
+	artifactPath := path.Join(rootDir, "gh-release-test")
+	provenanceFile := path.Join(artifactPath, "unittest.provenance")
+
+	ctx := context.Background()
+	owner, repo := "philips-labs", "slsa-provenance-action"
+	oauthClient := github.NewOAuth2Client(ctx, func() string { return githubToken })
+	client := github.NewReleaseClient(oauthClient)
+
+	releaseID, err := createGitHubRelease(
+		ctx,
+		client,
+		owner,
+		repo,
+		"v0.0.0-generate-test",
+		path.Join(rootDir, "bin", "slsa-provenance"),
+		path.Join(rootDir, "README.md"),
+	)
+	assert.NoError(err)
+
+	defer func() {
```

Comment:
Does this always happen, regardless of an error or failed test? Just need to make sure that releases are cleaned up, even if something goes wrong. 
If this is the case, please close this conversation.

Reply:
yup that is why we use defer. It is called as soon you exit the scope of the function

Reply:
Okay that's great. Then my assumption was correct.